internal/pkg/cli/project_show.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/describe"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// ShowProjectOpts contains the fields to collect for showing a project.
+type ShowProjectOpts struct {
+	shouldOutputJSON bool
+
+	storeSvc storeReader
+
+	w io.Writer
+
+	*GlobalOpts
+}
+
+// Validate returns an error if the values provided by the user are invalid.
+func (o *ShowProjectOpts) Validate() error {
+	if o.ProjectName() != "" {
+		_, err := o.storeSvc.GetProject(o.ProjectName())
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Ask asks for fields that are required but not passed in.
+func (o *ShowProjectOpts) Ask() error {
+	if err := o.askProject(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Execute shows the applications through the prompt.
+func (o *ShowProjectOpts) Execute() error {
+	projectToSerialize, err := o.retrieveData()
+	if err != nil {
+		return err
+	}
+	if o.shouldOutputJSON {
+		data, err := projectToSerialize.JSONString()
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(o.w, data)
+	} else {
+		fmt.Fprintf(o.w, projectToSerialize.HumanString())
+	}
+
+	return nil
+}
+
+func (o *ShowProjectOpts) retrieveData() (*describe.Project, error) {
+	proj, err := o.storeSvc.GetProject(o.ProjectName())
+	if err != nil {
+		return nil, fmt.Errorf("get project: %w", err)
+	}
+	envs, err := o.storeSvc.ListEnvironments(o.ProjectName())
+	if err != nil {
+		return nil, fmt.Errorf("list environment: %w", err)
+	}
+	apps, err := o.storeSvc.ListApplications(o.ProjectName())
+	if err != nil {
+		return nil, fmt.Errorf("list application: %w", err)
+	}
+	var envsToSerialize []*describe.Environment
+	for _, env := range envs {
+		envsToSerialize = append(envsToSerialize, &describe.Environment{
+			Name:      env.Name,
+			AccountID: env.AccountID,
+			Region:    env.Region,
+		})
+	}
+	var appsToSerialize []*describe.Application
+	for _, app := range apps {
+		appsToSerialize = append(appsToSerialize, &describe.Application{
+			Name: app.Name,
+			Type: app.Type,
+		})
+	}
+	return &describe.Project{
+		Name: proj.Name,
+		URI:  proj.Domain,
+		Envs: envsToSerialize,
+		Apps: appsToSerialize,
+	}, nil
+}
+
+func (o *ShowProjectOpts) askProject() error {
+	if o.ProjectName() != "" {
+		return nil
+	}
+	projNames, err := o.retrieveProjects()
+	if err != nil {
+		return err
+	}
+	if len(projNames) == 0 {
+		return fmt.Errorf("no project found: run %s or %s into your workspace please", color.HighlightCode("project init"), color.HighlightCode("cd"))
```

Comment:
Maybe something more colloquial:
no existing projects found. Run `init` to set one up.

Reply:
The problem with this one is: the number of available project is 0 can also mean that the user is not in a workspace (they might have several projects already). That's the reason why we rephrase the error message like this.

Reply:
How do you like this: no project found: run `project init` to set one up or `cd` into a workspace please